pkg/cmd/repo/autolink/view/http.go

```diff
@@ -0,0 +1,46 @@
+package view
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/cli/cli/v2/api"
+	"github.com/cli/cli/v2/internal/ghinstance"
+	"github.com/cli/cli/v2/internal/ghrepo"
+	"github.com/cli/cli/v2/pkg/cmd/repo/autolink/shared"
+)
+
+type AutolinkViewer struct {
+	HTTPClient *http.Client
+}
+
+func (a *AutolinkViewer) View(repo ghrepo.Interface, id string) (*shared.Autolink, error) {
+	path := fmt.Sprintf("repos/%s/%s/autolinks/%s", repo.RepoOwner(), repo.RepoName(), id)
+	url := ghinstance.RESTPrefix(repo.RepoHost()) + path
```

Comment:
Just noting it took me a hot minute to see `path` being reused later for error messaging if 404 arises.